sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollerFactory.java

```diff
@@ -0,0 +1,321 @@
+import com.azure.core.http.rest.Response;
+import com.azure.core.management.polling.PollResult;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollerFlux;
+import com.azure.core.util.polling.PollingContext;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Factory to create PollerFlux for Azure resource manager (ARM) long-running-operation (LRO).
+ */
+public final class PollerFactory {
+    private static final ClientLogger LOGGER = new ClientLogger(PollerFactory.class);
+    private static final LongRunningOperationStatus LRO_CANCELLED = LongRunningOperationStatus.fromString("Cancelled",
+        true);
+
+    /**
+     * Creates a PollerFlux with default ARM LRO init operation.
+     *
+     * @param serializerAdapter the serializer for any encoding and decoding
+     * @param pipeline the HttpPipeline for making any Http request (e.g. poll)
+     * @param pollResultType the type of the poll result, if no result is expecting then this should be Void.class
+     * @param finalResultType the type of the final result, if no result is expecting then this should be Void.class
+     * @param defaultPollInterval the default poll interval to use if service does not return retry-after
+     * @param lroInitMono the Mono on subscribe send the service request to initiate the long-running-operation
+     * @param <T> the type of poll result
+     * @param <U> the type of final result
+     * @return PollerFlux
+     */
+    public static <T, U> PollerFlux<PollResult<T>, U> create(
+        SerializerAdapter serializerAdapter,
+        HttpPipeline pipeline,
+        Type pollResultType,
+        Type finalResultType,
+        Duration defaultPollInterval,
+        Mono<Response<Flux<ByteBuffer>>> lroInitMono) {
+        Objects.requireNonNull(serializerAdapter, "'serializerAdapter' cannot be null.");
+        Objects.requireNonNull(pipeline, "'pipeline' cannot be null.");
+        Objects.requireNonNull(pollResultType, "'pollResultType' cannot be null.");
+        Objects.requireNonNull(finalResultType, "'finalResultType' cannot be null.");
+        Objects.requireNonNull(defaultPollInterval, "'defaultPollInterval' cannot be null.");
+        Objects.requireNonNull(lroInitMono, "'lroInitMono' cannot be null.");
+        Function<PollingContext<PollResult<T>>, Mono<PollResult<T>>> defaultLroInitOperation = context -> lroInitMono
```

Comment:
@yaohaizh what's your thought on this version of factory method that provide default lro init?

Reply:
My wish is that we could re-use the code in core for the deserialization (https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java#L496). It is quite complicated logic there (handle wired type, handle HEAD method, etc.). We probably not use that often, but it could be a hidden deviation of implementation.

I know it is really really hard to reuse core code... Just asking for advice.

Reply:
@anuchandy  sorry for delay. For this change, it is not resolve the operation status problem yet and the client have no way to determine the case of 200/201 cases.

Reply:
@anuchandy It seems we still need more change after the PR of https://github.com/Azure/azure-sdk-for-java/pull/7599 I'm okay with this change. @weidongxu-microsoft , could you help double check this PR from your side. Thanks.

Reply:
@yaohaizh you're right, https://github.com/Azure/azure-sdk-for-java/pull/7599 needs to be in for fully support 200/201 . At the moment team is doing some critical fixes in azure-core, I will discuss this with Jonathan and get back on merge plan and release of azure-core with initOp.